tests/test_declarations/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -1,5 +1,79 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
+exports[`angular_async.js 1`] = `
+beforeEach(async(() => {
+  // code
+}));
+
+afterAll(async(() => {
+  console.log('Hello');
+}));
+
+it('should create the app', async(() => {
+  //code
+}));
+
+it("does something really long and complicated so I have to write a very long name for the test", async(() => {
+  // code
+}));
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+beforeEach(async(() => {
+  // code
+}));
+
+afterAll(async(() => {
+  console.log("Hello");
+}));
+
+it("should create the app", async(() => {
+  //code
+}));
+
+it("does something really long and complicated so I have to write a very long name for the test", async(
+  () => {
+    // code
+  }
+));
+
+`;
+
+exports[`angular_async.js 2`] = `
+beforeEach(async(() => {
+  // code
+}));
+
+afterAll(async(() => {
+  console.log('Hello');
+}));
+
+it('should create the app', async(() => {
+  //code
+}));
+
+it("does something really long and complicated so I have to write a very long name for the test", async(() => {
+  // code
+}));
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+beforeEach(async(() => {
+  // code
+}));
+
+afterAll(async(() => {
+  console.log("Hello");
+}));
+
+it("should create the app", async(() => {
+  //code
+}));
+
+it("does something really long and complicated so I have to write a very long name for the test", async(
+  () => {
```

Comment:
We should keep them in one line in this case.

```js
it("does something really long and complicated so I have to write a very long name for the test", async(() => {
  // code
}));
```

You'll have to add one more condition here to match the inside `async()` since we have two possible linebreak in this case, the outside one (`it()`) and the inside one (`async()`), but we only disabled the outside one.

https://github.com/prettier/prettier/blob/01e8e2bb8a2035f6c37d94a30d63d8f9162d20f7/src/language-js/printer-estree.js#L962-L988

Reply:
Oh, I forgot we can add the condition in `isTestCall()`, it should be better.